src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -116,9 +121,10 @@ private BoundExpression BindDeconstruction(AssignmentExpressionSyntax node, Diag
                 var type = boundRHS.Type ?? voidType;
                 return new BoundDeconstructionAssignmentOperator(
                             node,
-                            DeconstructionVariablesAsTuple(node, checkedVariables, diagnostics, hasErrors: true),
+                            DeconstructionVariablesAsTuple(node, checkedVariables, resultIsUsed ? diagnostics : null, hasErrors: true),
```

Comment:
>DeconstructionVariablesAsTuple(node, checkedVariables, resultIsUsed ? diagnostics : null, hasErrors: true) [](start = 28, length = 106)

I am not comfortable with this change. It suppresses all diagnostics that could be reported by DeconstructionVariablesAsTuple function. It is very possible that might suppress to much in the future, or even does that already for some scenarios today. I think we should be very precise about what diagnostics we suppress and where.